crates/sim2h/src/lib.rs

```diff
@@ -192,17 +192,21 @@ impl Sim2h {
     /// we received some kind of error related to a stream/socket
     /// print some debugging and disconnect it
     fn priv_drop_connection_for_error(&mut self, uri: Lib3hUri, error: Sim2hError) {
-        error!(
-            "Transport error occurred on connection to {}: {:?}",
+        debug!(
+            "dropping connection to because of error {}: {:?}",
             uri, error,
         );
-        info!("Dropping connection to {} because of error", uri);
         self.disconnect(&uri);
     }
 
     /// if our connections sent us any data, process it
     fn priv_check_incoming_messages(&mut self) {
-        if let Ok((url, msg)) = self.msg_recv.try_recv() {
+        let len = self.msg_recv.len();
+        if len > 0 {
+            debug!("Handling {} incoming messages", len);
+        }
+        let v: Vec<_> = self.msg_recv.try_iter().collect();
```

Comment:
might it make sense to add a `take(100)` or so here, to make sure we're not starving other jobs on low cpu-count machines?

Reply:
i guess this is still our single thread bottleneck, so doesn't matter as much here

Reply:
I was guessing this would go away in what you are doing, so I just left it how I had it in the process-enhance branch